### PR TITLE
[RF] Put all temporary RooFit files in `/tmp/roofit` and not `/tmp` directly

### PR DESCRIPTION
**Pull request by guitargeek**
Some RooFit functions, in particular the ones related to
multiprocessing, created many files directly in the `/tmp` directory.
Some users and @hageboeck found that this was unnessecarily polluting
the `/tmp` directory and the files should rather go into a subdirectory.

This commit suggests to use the subdirectory `/tmp/roofit` for add
temporary RooFit files that get created during multiprocessing and
testing.

The path for temporary RooFit files can now be retrieved with a new
`RooFit::tmpPath()` function that is not exposted by the user but only
meant to be used by the RooFit implementations. The first time
`RooFit::tmpPath()` is called, it will check if the directory exists and
create it if it doesn't. This new function goes into a new package
`roofit/common` that is meant to be at the base of the RooFit dependency
graph, meaning all of `roofitcore`, `multiprocess`, and `roofitZMQ` can
depend on the `common` package to get the temporary files path.

### CODE REVIEW

File: roofit/multiprocess/src/Messenger.cxx [hunk 1]
```diff
@@ -63,9 +62,9 @@ Messenger::Messenger(const ProcessManager &process_manager)
 {
    sigemptyset(&ppoll_sigmask);
 
-   char addr_prefix_template[30] = "ipc:///tmp/%d_roofitMP";
-   char addr_prefix[50];
-   char addr[100];
+   auto makeAddrPrefix = [](pid_t pid) -> std::string {
+      return "ipc:///tmp/" + std::to_string(pid) + "_roofitMP";
```
Comment (lmoneta) [inline roofit/multiprocess/src/Messenger.cxx:66]: should not you use RooFit::tmpPath() here ?

**Review by lmoneta — APPROVED** [review]
LGTM ! Nice improvement!
Thank you Jonas for this. I have only a small observation, not sure if relevant

File: roofit/multiprocess/src/Messenger.cxx (hunk 1 above)
Comment (guitargeek) [reply roofit/multiprocess/src/Messenger.cxx:66]: Thanks for noticing, yes it should be used here! The change got lost when I factored my developments into two commits, thanks for spotting!

**Comment by guitargeek** [comment]
Just a small update moving the `#include <pthread.h>` from `BidirMMapPipe.cxx` to `BidirMMapPipe.h` (before, it was not noticed that it was missing because `#include "BidirMMapPipe.h"` was the final include in the `cxx` file and not the first one, so there was no header consistency check of `BidirMMapPipe.h` ).

**Pull request merged**